pkg/cmd/datastore/datastore.go

```diff
@@ -337,9 +387,9 @@ func newMySQLDatastore(opts Config) (datastore.Datastore, error) {
 		mysql.GCInterval(opts.GCInterval),
 		mysql.GCEnabled(!opts.ReadOnly),
 		mysql.GCMaxOperationTime(opts.GCMaxOperationTime),
-		mysql.ConnMaxIdleTime(opts.MaxIdleTime),
-		mysql.ConnMaxLifetime(opts.MaxLifetime),
-		mysql.MaxOpenConns(opts.MaxOpenConns),
+		mysql.MaxOpenConns(opts.ReadConnPool.MaxOpenConns),
```

Comment:
I think it could be argued some folks may consider setting the `WriteConnPool` parameter instead for those databases that do not support 2 conn pools. As a consequence, the user may think the flag is properly set, when it's in fact being ignored.

Ideally the application makes it difficult to make a mistake here.

Reply:
I agree -- especially since this PR doesn't split the pools for all of the datastores.

What's the precedent? We've got some other flags that are CRDB specific, right? What are we doing there?
Let's just be consistent in this PR and if we have to clean up that behavior, do everything at once in another PR.

Reply:
let's assume a user wants to move to use MySQL with the new connection flag, because the old ones are deprecated. They may think they have to set the `write` flags, when in fact they have to set the `read` flags. Does that make sense?

Reply:
Because MySQL doesn't use pgx, let's get this merged for now and we can work on getting parity to MySQL before the next release.